schematic_api/api/routes.py

```diff
@@ -40,10 +41,63 @@
 )
 from schematic.utils.general import entity_type_mapping
 from schematic.utils.schema_utils import get_property_label_from_display_name, DisplayLabelType
+from opentelemetry import trace
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+from opentelemetry.sdk.trace.export import BatchSpanProcessor, ConsoleSpanExporter, SimpleSpanProcessor, Span
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
 
 logger = logging.getLogger(__name__)
 logging.basicConfig(level=logging.DEBUG)
 
+trace.set_tracer_provider(
+    TracerProvider(
+        resource=Resource(attributes={SERVICE_NAME: "schematic-api"})
+    )
+)
+
+# borrowed from: https://github.com/Sage-Bionetworks/synapsePythonClient/blob/develop/tests/integration/conftest.py
+class FileSpanExporter(ConsoleSpanExporter):
+    """Create an exporter for OTEL data to a file."""
+
+    def __init__(self, file_path: str) -> None:
+        """Init with a path."""
+        self.file_path = file_path
+
+    def export(self, spans: List[Span]) -> None:
+        """Export the spans to the file."""
+        with open(self.file_path, "a", encoding="utf-8") as f:
+            for span in spans:
+                span_json_one_line = span.to_json().replace("\n", "") + "\n"
+                f.write(span_json_one_line)
+
+trace.get_tracer_provider().add_span_processor(BatchSpanProcessor(OTLPSpanExporter()))
+# processor = SimpleSpanProcessor(FileSpanExporter("otel_spans_schemati_api.json"))
+# trace.get_tracer_provider().add_span_processor(processor)
+tracer = trace.get_tracer("Schematic")
+
+def trace_function_params():
+    """capture all the parameters of API requests
+    """
+    def decorator(func):
+        """create a decorator
+        """
+        @wraps(func)
+        def wrapper(**kwargs: Any):
```

Comment:
Be careful only including `**kwargs` here. It's been a little bit since I've written a wrapper, but IIRC you should also be accepting `*args` too in the event you are decorating a function where non-kw arguments were used.

https://python.plainenglish.io/python-decorators-args-kwargs-oversimplified-5902a201717c

Reply:
Thanks for the comment! I added *args to allow any number of positional arguments to be used as well.